### ISSUE TITLE
[BUG]Applying cpu tweaks is not going to proceed

### ISSUE THREAD

**Issue by nabbiriri** [Teramanbr/TerabyteTweaker#4]
Thank you for making a good program.

However, as the title says, it does not proceed in the applying cpu tweaks stage.

I tested it in PC, notebook, and one(PC) was fine (but it did not proceed when re-applied), but the other one(notebook) continued to issue problems. Please check the notebook details as below.

Thanks

**Desktop (please complete the following information):**
 - OS: windows 10 lot enterprise Ltsc 
 - OS Version: 21H2
 - CPU/RAM: i5 5200u 2.2ghz/8gb

**Comment by Teramanbr** [comment]
Hello and thank you for your feedback!
Im sorry to hear you having problems with Terabyte Tweaker, It may have something to do about the core checking feature.
Here is how it works:
Terabyte Tweaker will check if your computer meets a standard for CPU or has HyperThreading turned ON, and if you only have 2 cores and no HyperThreading (Appears to be your case) it will try and skip most of the CPU tweaks and simply add some regedits for CPU improvement.
It may be a problem when creating a backup of your computer's actual Registry, which should be in the folder "C:\TT\TTRevert\ognic.reg" and if there is no such file in there, or the file is empty then the problem is the backup creation.

TLDR: Could you please check if the file "C:\TT\TTRevert\ognic.reg" exists and isn't blank? Thanks in advance. (also, always good to try using the tweaker again after rebooting your computer beforehand)

**Comment by nabbiriri** [comment]
Thank you for your quick response.
I am attaching two TT folders. I don't have the file you mentioned, but I
hope it helps.

Thanks.

2022년 9월 19일 (월) 오전 1:39, Teramanbr ***@***.***>님이 작성:

**Comment by Teramanbr** [comment]
Hello, just checked here and this file needs access so to see contents.
Please try and make it public, thanks.

**Comment by nabbiriri** [comment]
Please check it again.
thanks!

2022년 9월 20일 (화) 오전 4:30, Teramanbr ***@***.***>님이 작성:

**Comment by Teramanbr** [comment]
By checking the files, it is possible to see that the program indeed hasn't finished his actions.
It seems to be a problem specifically for your computer, are you able to check for the tweaks specifically?
[Test for CPU Tweaks.zip](https://github.com/Teramanbr/TerabyteTweaker/files/9602642/Test.for.CPU.Tweaks.zip)
This zip contains a batch file with only the CPU tweaks, could you test if it runs alone? If it doesn't then please tell me so i can divide the code in parts, which you can use to test to which point they run so we can map out where exactly the problem is.
Thanks in advance.

**Comment by nabbiriri** [comment]
I tried both PC and notebook, but it didn't work. Please let me know if you
need a log file like TT folder.

Thanks.

2022년 9월 20일 (화) 오전 7:14, Teramanbr ***@***.***>님이 작성:

**Comment by Teramanbr** [comment]
Divided the code in parts, could you tell me if any of these are able to run and if so, which aren't?
Download this [Test for Individual Tweaks](https://github.com/Teramanbr/TerabyteTweaker/files/9610676/Test.for.Individual.Tweaks.zip) and tell me, please.

Thanks

**Comment by nabbiriri** [comment]
Test 1, 2 Seems to be working properly
But when I run Test 3, the CMD pops up and then just disappears.
In conclusion, 1 and 2 are not problematic, but 3 seems to be problematic.

Thanks.

2022년 9월 21일 (수) 오전 4:19, Teramanbr ***@***.***>님이 작성:

**Comment by nabbiriri** [comment]
There's another problem I couldn't tell you.
The power plan was balanced, even if the Internet was connected. It may be
because it is not 100% complete, but please check it. (I submitted the TT
folder before, so you can check it with that.)

2022년 9월 21일 (수) 오전 6:24, 이세영 ***@***.***>님이 작성:

**Comment by Teramanbr** [comment]
The power plan not being changed appears to be a server issue, i was using the Power Plan directly from Hone's Discord and will from now on start using github as a host.

And since test 1 and 2 are working properly, seems it is not a problem on using wmic to check CPU Cores counting
I will check on this third test as it seems to not work properly on my computer aswell.

**Comment by Teramanbr** [comment]
Found out the reason for Terabyte Tweaker not being able to create the folder "C:\TT\TTRevert\ognic.reg" and also fixed it.

**Comment by Teramanbr** [comment]
Sorry for taking so long, seems like a problem on the cpu regedits, a part of the code that is easily removed without a big impact on peformance.

**Comment by nabbiriri** [comment]
I hope the CPU Tweak and the Power Plan problem will be solved so that I
can use it quickly. You had a hard time solving the problem.

Thank you.

2022년 9월 24일 (토) 오전 8:11, Teramanbr ***@***.***>님이 작성:

**Comment by Teramanbr** [comment]
Here you go, [this](https://github.com/Teramanbr/TerabyteTweaker/files/9638915/Terabyte.Tweaker.bat.zip) is the next update to Terabyte Tweaker and seems to be working correctly with me, but i want you to test it first to make it sure.

All of the updates inside of this file were already commited to github, only thing left is drafting the new release.

**Comment by nabbiriri** [comment]
I think the attached file is wrong.
When I extract it, a shortcut file appears.

2022년 9월 24일 (토) 오후 11:52, Teramanbr ***@***.***>님이 작성:

**Comment by Teramanbr** [comment]
Oh my god, im so sorry!
I normally have a shortcut to the tweaker on my desktop and i just copypasted it when i got it working.
[Here you go!](https://github.com/Teramanbr/TerabyteTweaker/files/9639816/Terabyte.Tweaker.zip)
Sorry again and thanks for your patience!

**Comment by nabbiriri** [comment]
TT(220925).zip
<https://drive.google.com/file/d/1OyCqSV9-qDFtbBZ6ShiPUCXOcEJClbs8/view?usp=drive_web>
I've only done it on my laptop, but unfortunately it still doesn't work at
45%.
Please refer to the attachment for specifications of the laptop.
(Since it's a laptop, I think it might be because it has an internal and
external graphics card.)

2022년 9월 25일 (일) 오전 9:54, Teramanbr ***@***.***>님이 작성:

**Comment by Teramanbr** [comment]
Just like before, access denied on google drive.

**Comment by nabbiriri** [comment]
I shared it, so please check it out

2022년 9월 25일 (일) 오전 11:10, Teramanbr ***@***.***>님이 작성:

**Comment by nabbiriri** [comment]
I removed the CPU Tweak part from the bat file and ran it, and it ran
without any problems from beginning to end. I think it's definitely a CPU
Tweak problem, so I hope it's fixed.
 Terabyte Tweaker.bat
<https://drive.google.com/file/d/1maELwxkKntpYIlCkYp0IUTIzKrMnrCMG/view?usp=drive_web>


2022년 9월 25일 (일) 오전 11:58, 이세영 ***@***.***>님이 작성:

**Comment by Teramanbr** [comment]
This problem seems really hard to fix but i will try my best, apparently, it happens specifically on your computer.
I have also tested this "beta" version on other computer and it seems to work properly, but i will run more tests and will release an major update to the tweaker really soon.

**Comment by Teramanbr** [comment]
Just released a new stable version to Terabyte Tweaker, could you check it out? Here is the [Download](https://github.com/Teramanbr/TerabyteTweaker/releases/download/v1.13.0/Terabyte.Tweaker.bat).
Thanks.

**Comment by Teramanbr** [comment]
Just tested with a friend's notebook and apparently it happens on his too, maybe it is a problem specifically on notebooks.
The only way to got past this point apparently is clicking CTRL+C and then N to cancel the actual procedure and then continue using the rest of the program.
This should be a temporary fix, while i don't find out exactly the problem.

**Comment by nabbiriri** [comment]
I'm late, but I'm attaching the TT folder for PC. I hope it helps you solve
the problem.
Thanks.
 TT(PC).zip
<https://drive.google.com/file/d/1ENq0jMHoecwTrtiJO3RKUNT8QGJCz1XE/view?usp=drive_web>


2022년 9월 26일 (월) 오전 3:39, Teramanbr ***@***.***>님이 작성:

**Comment by andreykolesnikov** [comment]
Hey, I have the same issues. And I have not empty ognic.reg file in TT folder. PC on Windows 10

**Comment by maurye80** [comment]
same here with a Lenovo Laptop ideapad windows 10 And I have ognic.reg file too.

a big thank you anyway for this great program, it works really well apart from this small problem. I am sure you will quickly find the solution. you're a master, respect!

**Comment by Teramanbr** [comment]
Program seems to run fully, apart for the CPU tweaks and seems to be creating a loop specifically on laptops and not in desktops. 
At least this is my main suspiction right now, this will be really hard to test, mainly because i don't have a laptop.
Will try my best though.

**Comment by Teramanbr** [comment]
Finding a way to emulate a laptop specifically has proven itself to be an incredibily hard task, i may have to borrow a friend's PC soon just to be able to see what is happening properly.

**Comment by Teramanbr** [comment]
I think i know what we can do, can you remove the first line of code "@echo off" and run the program? This will show specifically what is happening and which lines of code are being run.
If there is a loop then please send the entire loop to me, if there is a specific command where it just stops then send the command alone and I will try and fix it.
It should look like this, it will still work as if nothing was happening but the interface will hurt eyes a little bit: 
![image](https://user-images.githubusercontent.com/71410827/193428438-d16d19d5-c2eb-4c75-a6bd-04849d4cb56b.png)

**Comment by nabbiriri** [comment]
I attach the screen.

2022년 10월 2일 (일) 오전 6:11, Teramanbr ***@***.***>님이 작성:

**Comment by Teramanbr** [comment]
I think you forgot to paste the screenshot.

**Comment by nabbiriri** [comment]
2022-10-02 10 30 21.png
<https://drive.google.com/file/d/1Nkxj3D8LDNfgl4CLYFgXyU2-Td-2QkWY/view?usp=drive_web>


2022년 10월 3일 (월) 오후 6:47, Teramanbr ***@***.***>님이 작성:

**Comment by Teramanbr** [comment]
Here you go, i think [this should fix it](https://raw.githubusercontent.com/Teramanbr/TerabyteTweaker/main/src/Terabyte%20Tweaker.bat).
Apparently, the problem is on getting to the next part of the code. I shortened it a little and it should work now.

**Comment by nabbiriri** [comment]
2022-10-05 05 56 42.png
<https://drive.google.com/file/d/1wx2Y86u_OLD5ihcP-hS98F4cf7wWL3ar/view?usp=drive_web>
Unfortunately, it is still the same. I'm attaching a screen capture.

2022년 10월 5일 (수) 오전 3:09, Teramanbr ***@***.***>님이 작성:

**Comment by izzatsksk** [comment]
Where do i find the restore point? It seems when i use this, my fingerprint scanner stopped working

**Comment by Teramanbr** [comment]
Tried again and I think this should work now, after adding some more measures to prevent it from suddenly stopping: 
[Download](https://github.com/Teramanbr/TerabyteTweaker/files/9735563/Terabyte.Tweaker.zip)

**Comment by nabbiriri** [comment]
It's still not working, but I think it's almost solved.
 TT.zip
<https://drive.google.com/file/d/1lQk27_aOppWsWLVARwY9Bbu0FBWzWClY/view?usp=drive_web>

 2022-10-08 08 16 37.png
<https://drive.google.com/file/d/1B2Flhf5bxJssJiP5aaDxNp4yVPSRLcgH/view?usp=drive_web>


2022년 10월 8일 (토) 오전 1:26, Teramanbr ***@***.***>님이 작성:

**Comment by Teramanbr** [comment]
Working on another attempt, will be ready by tomorrow!
Sorry for taking so long to answer, school's got me really busy.

**Comment by Teramanbr** [comment]
[And here is another attempt.](https://github.com/Teramanbr/TerabyteTweaker/files/9769196/Terabyte.Tweaker.zip)
Hope it works now, i really don't know what else to change to make it stop happening, and can't seem to find a command for timeoutting other commands after an amount of time.

**Comment by nabbiriri** [comment]
ㅠ.ㅠ... not working...
 2022-10-13 23 26 05.png
<https://drive.google.com/file/d/1_gVYVOqkA05gv7nnvoouyLs_VQknpSDH/view?usp=drive_web>

 TT.zip
<https://drive.google.com/file/d/1tQHHEPDXr-bzlcLwZDcPRozOYc_S3fQv/view?usp=drive_web>


2022년 10월 13일 (목) 오전 8:10, Teramanbr ***@***.***>님이 작성:

**Comment by Teramanbr** [comment]
Yeah, i am taking a while to learn how to use PowerShell, will update as soon as I'm able actually understand how to use it.

**Comment by Teramanbr** [comment]
I took a lot of time but think I finally did it.
Please, try out [The Latest Release](https://github.com/Teramanbr/TerabyteTweaker/releases/latest), It should work normally now.

**Comment by maurye80** [comment]
Everything seems to be working correctly since your update, the process ended without crashing or blocking! A big thank you and bravo for your determination, respect

**Comment by nabbiriri** [comment]
Finally, it works well on my desktop and notebook! I thank you for your
effort.

**Comment by Teramanbr** [comment]
Thank you all for waiting, now if you excuse me, I gotta lay down in fetal position for some hours.
